src/topology/sets/opens.lean

```diff
@@ -18,75 +19,100 @@ We define the subtype of open sets in a topological space.
 
 ## Main Definitions
 
+### Bundled open sets
+
 - `opens α` is the type of open subsets of a topological space `α`.
+- `opens.is_basis` is a predicate saying that a set of `opens`s form a topological basis.
+- `opens.comap`: preimage of an open set under a continuous map as a `frame_hom`.
+- `homeomorph.opens_congr`: order-preserving equivalence between open sets in the domain and the
+  codomain of a homeomorphism.
+
+### Bundled open neighborhoods
+
 - `open_nhds_of x` is the type of open subsets of a topological space `α` containing `x : α`.
+- `open_nhds_of.comap f x U` is the preimage of open neighborhood `U` of `f x` under `f : C(α, β)`.
+
+## Main results
+
+We define order structures on both `opens α` (`complete_structure`, `frame`) and `open_nhds_of x`
+(`order_top`, `distrib_lattice`).
 -/
 
 open filter function order set
+open_locale topology
 
 variables {ι α β γ : Type*} [topological_space α] [topological_space β] [topological_space γ]
 
 namespace topological_space
+
 variable (α)
+
 /-- The type of open subsets of a topological space. -/
-def opens := {s : set α // is_open s}
+structure opens :=
+(carrier : set α)
+(is_open' : is_open carrier)
 
 variable {α}
+
 namespace opens
-instance : has_coe (opens α) (set α) := { coe := subtype.val }
 
-lemma val_eq_coe (U : opens α) : U.1 = ↑U := rfl
+instance : set_like (opens α) α :=
+{ coe := opens.carrier,
+  coe_injective' := λ ⟨_, _⟩ ⟨_, _⟩ _, by congr; assumption }
+
+instance : can_lift (set α) (opens α) coe is_open :=
+⟨λ s h, ⟨⟨s, h⟩, rfl⟩⟩
+
+lemma «forall» {p : opens α → Prop} : (∀ U, p U) ↔ ∀ (U : set α) (hU : is_open U), p ⟨U, hU⟩ :=
+⟨λ h _ _, h _, λ h ⟨U, hU⟩, h _ _⟩
```

Comment:
I think you can prove this with `set_like.forall`

Reply:
No, I can't. This lemma is about a different type of forall.